app/models/page.rb

```diff
@@ -1,2 +1,15 @@
 class Page < ApplicationRecord
+  has_one :public_page
+
+  validates :page_type, presence: true
+  validates :content, uniqueness: true
+  validates :page_type, uniqueness: true
+
+  after_save do
+    if published
+      public_page = PublicPage.where(page_id: id).first_or_initialize
+      public_page.content = content
+      public_page.save
```

Comment:
How about:

```ruby
PublicPage.find_or_create_by(page_id: id) do |public_page|
  public_page.content = content
end
```

I didn't test it ⬆️

Reply:
@lucascaton , thanks for the suggestion . I tried this, however, this doesn't , work when we want to update an existing record.